### DISCUSSION TITLE
Does update re-create the complete index?

### DISCUSSION THREAD

**Discussion by nw-mjl** [Jeroen-G/Explorer#199] [General]
I am evaluating into using scout with ES for search and different drivers. 
I did not find much on how the update command works, but based on the index alias docs, it seems like it recreates the complete index every time? Is that the case?

Is there a way to only add the new rows to the index?.

We are indexing many millions of documents so I am will need to test how this will affect updating the indexes.   

Thanks for any info you can provide.

**Comment by Jeroen-G** [comment]
My view, and thus that of this package, is that recreating indices is better (supported) than updating the same index like you would with a database.
ElasticSeach has aliases for this (see also docs of this package). So using the update command will create a new index in the background and when done switch the alias from the old to the new.
If you really want to add new documents to an index instead of using an alias you will have to write a new adapter for that part of the ES API because the one in this package does not have that functionality (and probably never will).
